Framework/Algorithms/src/CalculateCountRate.cpp

```diff
@@ -384,9 +385,18 @@ void CalculateCountRate::setOutLogParameters(const DataObjects::EventWorkspace_s
         if (!m_tmpLogHolder) {
           m_tmpLogHolder = std::unique_ptr<Kernel::TimeSeriesProperty<double>>(m_pNormalizationLog->clone());
         }
-        m_tmpLogHolder->filterByTime(runTMin, runTMax);
-        m_pNormalizationLog = m_tmpLogHolder.get();
-        m_numLogSteps = m_pNormalizationLog->realSize();
+        Kernel::TimeROI roi(runTMin, runTMax);
+        if (Kernel::FilteredTimeSeriesProperty<double> *filteredLog =
+                dynamic_cast<Kernel::FilteredTimeSeriesProperty<double> *>(m_tmpLogHolder.get())) {
+          filteredLog->filterWith(roi);
```

Comment:
(rant) this `filterWith` name is unfortunate. I thought the function was actually filtering the time series, but actually is just replacing its internal filter.

Reply:
The name is even in the tutorials. We're stuck on this one.